### PR TITLE
[integrations][okta] - Added support for Oauth2 for okta.system data stream

### LINKED ISSUE #6974: [Meta][Okta] Support OAuth 2.0 for Integration
The current [Okta Logs Integration](https://docs.elastic.co/integrations/okta) supports Basic Auth using an API key to consume Okta System Logs.

Okta [supports OAuth 2.0](https://developer.okta.com/docs/guides/implement-oauth-for-okta/main/) for [scoped API access to System Logs](https://developer.okta.com/docs/api/oauth2/#okta-admin-management) which is a preferable authentication mechanism for some / most users.

This issue is a request to add OAuth as an authentication method for the Okta integration.

### PR DESCRIPTION
**Pull request by ShourieG**
## Type of change
- Enhancement

## What does this PR do?
This PR adds OAuth2 support for Okta System Logs data stream in the Okta integration.

## Note
This PR can only be merged after the public release of 8.10 because of internal dependencies and features only available in 8.10.  **System tests will fail until 8.10 changes are available in the CI pipeline.**

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues
- Closes #6974

## Screenshots

Screenshot 2023-09-11 at 12 50 49 PM

### CODE REVIEW

File: packages/okta/_dev/build/docs/README.md [hunk 1]
```diff
@@ -8,6 +8,27 @@ The Okta integration collects events from the Okta API, specifically reading fro
...
 The Okta System Log records system events related to your organization in order to provide an audit trail that can be used to understand platform activity and to diagnose problems. This module is implemented using the httpjson input and is configured to paginate through the logs while honoring any rate-limiting headers sent by Okta.
 
+## Types Of Authentication
+### API Key
+In this type of authentication, we only require an API Key for authenticating the client and polling for Okta System Logs.
+
+### Oauth2
+**In this type of authentication, we require the following information:**
+1. Your Okta domain URL. [ Example: https://dev-123456.okta.com ]
+2. Your Okta service app Client ID.]
```
Comment (bhapas) [inline packages/okta/_dev/build/docs/README.md:18]:
```suggestion
2. Your Okta service app Client ID.
```

**Review by bhapas — COMMENTED** [review]
LGTM.

Is there a way to document the common pitfalls regarding the config of OAuth2 on Okta server. 
Lately, there were few issues with other products that customers faced regarding misconfiguration of the OAuth servers and leading to integration issues.

**Comment by ShourieG** [comment]
> LGTM.
> 
> Is there a way to document the common pitfalls regarding the config of OAuth2 on Okta server. Lately, there were few issues with other products that customers faced regarding misconfiguration of the OAuth servers and leading to integration issues.

@bhapas The main issue I see is different oauth2 implementations require specific steps from users to get things working, specially in the case of machine to machine communication. The implementation of oauth2 followed by okta here deviates from standard oauth2 conventions making a single document covering pitfalls of oauth2 as a whole somewhat difficult. It needs to be case by case in my opinion.

**Comment by bhapas** [comment]
> > LGTM.
> > Is there a way to document the common pitfalls regarding the config of OAuth2 on Okta server. Lately, there were few issues with other products that customers faced regarding misconfiguration of the OAuth servers and leading to integration issues.
> 
> @bhapas The main issue I see is different oauth2 implementations require specific steps from users to get things working, specially in the case of machine to machine communication. The implementation of oauth2 followed by okta here deviates from standard oauth2 conventions making a single document covering pitfalls of oauth2 as a whole somewhat difficult. It needs to be a case by case basis in my opinion.

@ShourieG Yeah. I meant the OAuth2 configuration for Okta servers , not in general.

**Comment by ShourieG** [comment]
> > > LGTM.
> > > Is there a way to document the common pitfalls regarding the config of OAuth2 on Okta server. Lately, there were few issues with other products that customers faced regarding misconfiguration of the OAuth servers and leading to integration issues.
> > 
> > 
> > @bhapas The main issue I see is different oauth2 implementations require specific steps from users to get things working, specially in the case of machine to machine communication. The implementation of oauth2 followed by okta here deviates from standard oauth2 conventions making a single document covering pitfalls of oauth2 as a whole somewhat difficult. It needs to be a case by case basis in my opinion.
> 
> @ShourieG Yeah. I meant the OAuth2 configuration for Okta servers , not in general.

@bhapas Yea we will take up this discussion point in the team meet and figure out a better way for these documentations.

**Review by bhapas — APPROVED** [review]
LGTM.

Can you add some screenshots for the changed UI

**Pull request merged**